src/components/select/select.template.tsx

```diff
@@ -210,3 +213,16 @@ function getSelectedText(value: Option | Option[]): string {
 
     return value.text;
 }
+
+function getSelectedTextIcon(value: any) {
+    if (!value) {
```

Comment:
```suggestion
    if (!value || !value.icon) {
```

That ought to fix the empty icon when there's no name. But I can test it while I review. Gotta do another review first though.